src/index.ts

```diff
@@ -103,30 +86,44 @@ export default function cssInjectedByJsPlugin({
                 });
             }
 
-            const cssInjectionCode = await buildCSSInjectionCode({
-                cssToInject: typeof preRenderCSSCode == 'function' ? preRenderCSSCode(cssToInject) : cssToInject,
-                styleId,
-                injectCode,
-                injectCodeFunction,
-                useStrictCSP,
-            });
-
             if (jsAssetTargets.length == 0) {
                 throw new Error(
                     'Unable to locate the JavaScript asset for adding the CSS injection code. It is recommended to review your configurations.'
                 );
             }
 
-            jsAssetTargets.forEach((jsAsset) => {
+            for (const jsAsset of jsAssetTargets) {
+                // @ts-ignore
+                const cssNamesSet: Set<string> = jsAsset.viteMetadata.importedCss;
+                let cssToInject: string = '';
+                cssNamesSet.forEach((cssName) => {
+                    cssToInject += String(extractCssCode(bundle[cssName] as OutputAsset));
+                });
```

Comment:
I think you can delete directly here the OutputAsset once the code is extracted

Reply:
Some CSS bundles can be used in several JS assets, if we delete the bundle here other JS assets can't use the CSS.  
For example, I have a `Button` that uses `Spinner` for loading, the button should use both `spinner.css` and `button.css` and the plugin will remove `spinner.css` from bundles and while vite is bundling the spinner, `spinner.css` is removed.

Reply:
You're right, sorry I didn't think about it